proposals/flexible-vectors/FlexibleVectors.md

```diff
@@ -0,0 +1,327 @@
+Flexible vectors overview
+=========================
+
+The goal of this proposal is to provide flexible vector instructions for
+WebAssembly as a way to bridge the gap between existing SIMD instruction sets
+available on various platforms. `simd128` proposal already identified
+operations that would commonly work on platforms that are important to
+WebAssembly, this proposal is attempting to extend the same operations to work
+with variable vector lengths.
+
+## Types
+
+Proposal introduces the following vector types:
+
+- `vec.v8` : 8-bit lanes
+- `vec.v16`: 16-bit lanes
+- `vec.v32`: 32-bit lanes
+- `vec.v64`: 64-bit lanes
+
+### Lane division interpretation
+
+In semantic pseudocode `S` is the particular vector type, `S.LaneBits` is the
+size of the lane in bits, `S.Lanes` is the number of lanes, which is dynamic.
+
+|    S      | S.LaneBits |
+|-----------|-----------:|
+| `vec.v8`  |          8 |
+| `vec.v16` |         16 |
+| `vec.v32` |         32 |
+| `vec.v64` |         64 |
+
+### Restrictions
+
+Lane values are intended to be handled exactly like in `simd128` proposal, with
+the following differences applying to overall types:
+
+- Runtime sets maximum vector length for every type
+- Number of lanes is set separately for different lane sizes
```

Comment:
Are runtimes free to use different maximum vector length (in bits) for different types? Good to clarify.

Reply:
Potentially yes, granularity is what the exposed get/set length functions are - current draft differentiates between lane sizes, while could be possible to add the data type to the mix if necessary. The overall philosophy is to let runtimes interact with get/set instruction to make the appropriate decisions.

I am not sure if this would be ever used or if supporting different lengths is useful.

Reply:
That is now part of #6